common/post-processing-worker-filter.h

```diff
@@ -22,17 +22,31 @@ class post_processing_worker_filter : public post_processing_filter
     }
     ~post_processing_worker_filter()
     {
-        _alive = false;
-        if( _worker.joinable() )
-            _worker.join();
+        if (_alive)
+        {
+            _alive = false;
+            if (_worker.joinable())
+            {
+                if (std::this_thread::get_id() != _worker.get_id())
+                {
+                    _worker.join();
+                }
+                else {
+                    _worker.detach();
+                }
+            }
+        }
+        
     }
 
 public:
     void start( rs2::subdevice_model & model ) override
     {
-        post_processing_filter::start( model );
-        _worker = std::thread( [&]()
-        {
+        post_processing_filter::start(model);
+        auto p = model.shared_from_this();
```

Comment:
Please use clearer names than `p` and `pp` :)

Reply:
done